src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -191,20 +191,16 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions)
             if (Utility.IsSupportedRuntime(_workerRuntime, _workerConfigs))
             {
                 State = FunctionDispatcherState.Initializing;
-                Dictionary<string, TaskCompletionSource<ILanguageWorkerChannel>> webhostLanguageWorkerChannels = _webHostLanguageWorkerChannelManager.GetChannels(_workerRuntime);
-                if (webhostLanguageWorkerChannels != null)
+                IEnumerable<ILanguageWorkerChannel> initializedChannels = _webHostLanguageWorkerChannelManager.GetChannels(_workerRuntime);
```

Comment:
same here. We need to keep TaskCompletionSource to handle race conditions where specialization happens before process startup finishes

Reply:
I moved that logic inside of "LanguageWorkerChannel" directly, so everything happens in the correct order, but the order is just maintained in "LanguageWorkerChannel" itself without having to make sure that its callers are preserving order in a very specific way